pkg/cloud/api/path.go

```diff
@@ -71,6 +87,33 @@ func (p Path) Equal(other Path) bool {
 	return true
 }
 
+// Match the given path, interpreting wildcard matching for the comparison. Use
+// Equal() to compare Paths for absolute equality (no wildcard interpretation).
+func (p Path) Match(other Path) bool {
+	if len(p) != len(other) {
+		return false
+	}
+	for i := range p {
+		// Wildcard match for slices and maps indices
+		if isAnySliceIndexElement(p[i]) && isSliceIndex(other[i]) {
+			continue
+		}
+		if isAnySliceIndexElement(other[i]) && isSliceIndex(p[i]) {
+			continue
+		}
+		if isAnyMapIndexElement(p[i]) && isMapIndex(other[i]) {
+			continue
+		}
+		if isAnyMapIndexElement(other[i]) && isMapIndex(p[i]) {
+			continue
+		}
+		if p[i] != other[i] {
+			return false
+		}
+	}
+	return true
+}
+
 // HasPrefix returns true if prefix is the prefix of this path.
 func (p Path) HasPrefix(prefix Path) bool {
```

Comment:
Let's add to the docstring

// HasPrefix returns true if prefix matches this path. Prefix
// uses Match() semantics for wildcards.